src/wp-includes/widgets/class-wp-widget-media-image.php

```diff
@@ -239,14 +239,31 @@ public function render_media( $instance ) {
 				$instance['height'] = '';
 			}
 
-			$image = sprintf(
-				'<img class="%1$s" src="%2$s" alt="%3$s" width="%4$s" height="%5$s" />',
-				esc_attr( $classes ),
-				esc_url( $instance['url'] ),
-				esc_attr( $instance['alt'] ),
-				esc_attr( $instance['width'] ),
-				esc_attr( $instance['height'] )
+			$attr = array(
+				'class'    => $classes,
+				'src'      => $instance['url'],
+				'alt'      => $instance['alt'],
+				'width'    => $instance['width'],
+				'height'   => $instance['height'],
+				'decoding' => 'async',
 			);
+
+			$loading_optimization_attr = wp_get_loading_optimization_attributes(
+				'img',
+				$attr,
+				'widget_media_image'
+			);
+
+			$attr = array_merge( $attr, $loading_optimization_attr );
+
+			$attr  = array_map( 'esc_attr', $attr );
```

Comment:
Just for your information, in the `$attr` array, the `src` attribute should use `esc_url` instead of `esc_attr`. It's worth noting that the `wp_get_attachment_image()` function uses the same code for escaping the `src` attribute.

Reply:
I just found that `esc_url()` must not be used because it will reject `data:` URLs: https://github.com/WordPress/wordpress-develop/pull/4786/files#r1253728836